src/utils/filterTodo.ts

```diff
@@ -0,0 +1,23 @@
+import { TODO_FILTER_STATUS } from '../constants/TodoFilter';
+import { ITodoFilter, Todo } from '../types/Todo';
+
+export const getTodoFilterStatus = (status = '') => {
+  if (status in TODO_FILTER_STATUS) {
+    return TODO_FILTER_STATUS[status as keyof typeof TODO_FILTER_STATUS];
+  }
+  return undefined;
+};
+
+export const getFiltredTodo = (
+  todos: Todo[],
+  { status, filter }: ITodoFilter,
+) => {
+  const todoStatus = getTodoFilterStatus(status);
```

Comment:
```suggestion
   let filteredTodos = [...todos];

  filteredTodos = filteredTodos.filter(({ completed }) => {
    switch (status) {
      case Filter_Statuses.Active:
        return !completed;
      case Filter_Statuses.Completed:
        return completed;
      default:
        return true;
    }
  });
```
under this u can do an additional check on the query

Reply:
yes,but i need to use two cycle,i thought that use one cycle is less expensive